File: problems/3000/Solution.cs
public class Solution {
    public int AreaOfMaxDiagonal(int[][] dimensions) {
        int bestDiag = 0, bestArea = 0;
        foreach (int[] d in dimensions) {
            int length = d[0], width = d[1];
            int diag = length * length + width * width;
            int area = length * width;
            if (diag > bestDiag || (diag == bestDiag && area > bestArea)) {
                bestDiag = diag;
                bestArea = area;
            }
        }
        return bestArea;
    }
}

File: problems/3000/Solution.kt
class Solution {
    fun areaOfMaxDiagonal(dimensions: Array<IntArray>): Int {
        var bestDiag = 0
        var bestArea = 0
        for (d in dimensions) {
            val length = d[0]
            val width = d[1]
            val diag = length * length + width * width
            val area = length * width
            if (diag > bestDiag || (diag == bestDiag && area > bestArea)) {
                bestDiag = diag
                bestArea = area
            }
        }
        return bestArea
    }
}
